### PR TITLE
Feature/glorious refactor

### PR DESCRIPTION
**Pull request by dsellarsnr**

### CODE REVIEW

File: shared/agent/src/system/decorators/lsp.ts [hunk 1]
```diff
@@ -103,9 +94,8 @@ export function registerProviders(
 	if (clear) {
 		providerRegistry.clear();
 	}
-	const mergedProviders = mergeLocalProviders(providers);
-	for (const providerId in mergedProviders) {
-		const provider = mergedProviders[providerId];
+	for (const providerId in providers) {
+		const provider = providers[providerId];
```
Comment (bcanzanella) [inline shared/agent/src/system/decorators/lsp.ts:98]: mergedProviders is no longer a thing?

File: shared/agent/src/session.ts [hunk 2]
```diff
@@ -296,8 +286,6 @@ export class CodeStreamSession {
 
 		Container.initialize(agent, this);
 
-		registerProviders(PROVIDERS_TO_REGISTER_BEFORE_SIGNIN, this);
```
Comment (bcanzanella) [inline shared/agent/src/session.ts:299]: not needed?

**Comment by bcanzanella** [comment]
couple comments, otherwise LGTM. Let's get @marcelofarias to take a look too.

File: shared/agent/src/system/decorators/lsp.ts (hunk 1 above)
Comment (dsellarsnr) [reply shared/agent/src/system/decorators/lsp.ts:98]: mergedProviders was a hack to get nrVulnerability.ts to work as a lsp provider / extends ThirdPartyProviderBase, it is indeed no longer a thing

File: shared/agent/src/session.ts (hunk 2 above)
Comment (dsellarsnr) [reply shared/agent/src/session.ts:299]: No longer a lsp provider / extends ThirdPartyProviderBase - replaced by injectNR in SessionServiceContainer

**Review by marcelofarias — APPROVED** [review]

**Pull request merged**